src/main/java/me/dancedog/rewardclaim/RewardListener.java

```diff
@@ -86,12 +84,13 @@ public void onClientTick(ClientTickEvent event) {
    */
   @SubscribeEvent
   public void onChatReceived(ClientChatReceivedEvent event) {
-    Matcher chatMatcher = REWARD_MESSAGE_PATTERN.matcher(event.message.getFormattedText());
-    if (chatMatcher.find()) {
+    boolean chatMatcher = event.message.getUnformattedText().contains("Click the link to visit our website and claim your reward: https://rewards.hypixel.net/claim-reward/");
```

Comment:
Completely unrelated to making the mod work again, but since there's always `rewards.hypixel.net` in the string wouldn't it be better to just search for that part so it works for multiple locales? Also if the message is changed by Hypixel it will probably keep working since the url base won't change.
Or does that link appear in other messages?

Reply:
I like the use of `getUnformattedText`, but a potential security issue is that someone could send a message containing that format, e.g...
`[MVP+] ImAPlayer: Click the link to visit our website and... (etc)`
and the mod would pick that up and try to open it.
```suggestion
    boolean chatMatcher = event.message.getUnformattedText().startsWith("Click the link to visit our website and claim your reward: https://rewards.hypixel.net/claim-reward/");
```
or 
```suggestion
    boolean chatMatcher = event.message.getUnformattedText().matches("Click the link to visit our website and claim your reward: https:\\/\\/rewards\\.hypixel\\.net\\/claim-reward\\/([\\w]+)");
```
Should also rename the boolean to `isRewardMessage` or something

Reply:
```suggestion
    boolean chatMatcher = event.message.getUnformattedText().startsWith("\nClick the link to visit our website and claim your reward: https://rewards.hypixel.net/claim-reward/");
```
Just checked and for the startsWith and matches there is a newline at the beginning of the message that would need to be added. It may be the reason why it started not to work. Just adding \n at the beginning of the literal would makes it work.